Directory.Build.targets

```diff
@@ -0,0 +1,31 @@
+<Project>
+    <Target
+            Name="PublishSamples"
+            Condition="'$(IsSample)' == 'true'">
+        <PropertyGroup>
+            <ArtifactsRoot>$(MSBuildThisFileDirectory)/artifacts/</ArtifactsRoot>
+            <_IsMultiTargetTfm Condition="'$(TargetFrameworks)' != ''and '$(TargetFramework)' == ''">true</_IsMultiTargetTfm>
+            <_IsSingleTargetTfmNoRids Condition="'$(TargetFramework)' != '' and '$(RuntimeIdentifiers)' == ''">true</_IsSingleTargetTfmNoRids>
+            <_IsSingleTargetAndMultiRid Condition="'$(TargetFramework)' != '' and '$(RuntimeIdentifiers)' != ''">true</_IsSingleTargetAndMultiRid>
+        </PropertyGroup>
+        <!-- TFMs but no TF -> multitarget, making samples for each TFM -->
+        <ItemGroup Condition="'$(_IsMultiTargetTfm)' == 'true'">
+            <_TFMItems Include="$(TargetFrameworks)" />
+            <_SingleSamplePublish
+                    Include="$(MSBuildProjectFullPath)"
+                    AdditionalProperties="TargetFramework=%(_TFMItems.Identity);PublishDir=$(ArtifactsRoot)$(MSBuildProjectName)/%(_TFMItems.Identity)" />
+        </ItemGroup>
```

Comment:
If we have multiple TFMs, then we want to Publish the project multiple times - once for each TFM. And since we need to deduplicate the published outputs, we add the TFM onto the outputs directory here. This logic is something very much like what the SDK might do - but for some publish targets like Containers, more customization might be needed (as well as some kind of post-TFMs-published hook).